client/my-sites/checkout/upsell-nudge/purchase-modal/index.tsx

```diff
@@ -142,10 +144,10 @@ export default function PurchaseModalWrapper( props: PurchaseModalProps ) {
 		]
 	);
 
-	useEffect( () => {
-		let isUpdatingCart = false;
+	const [ requestSent, setRequestSent ] = useState( false );
```

Comment:
This is much better than ignoring the exhausted deps! 👍 

Minor and probably not very important in this particular case, but since we don't need a rerender when this is updated, and we want to know immediately if the request has been sent, using `useRef` here might be a better choice.

Reply:
Thanks for the suggestion! Since the PR is functioning, I am shipping it now. I'll handle all code-related follow-ups (this one and the one mentioned above by Jeremy) in a separate PR.